core/services/relay/ethereum/ethereum.go

```diff
@@ -0,0 +1,211 @@
+package ethereum
+
+import (
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	offchain_aggregator_wrapper "github.com/smartcontractkit/chainlink/core/internal/gethwrappers2/generated/offchainaggregator"
+	txm "github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/relay"
+	"github.com/smartcontractkit/libocr/gethwrappers2/ocr2aggregator"
+	"github.com/smartcontractkit/libocr/offchainreporting2/chains/evmutil"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median/evmreportcodec"
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/service"
+	ocr2 "github.com/smartcontractkit/chainlink/core/services/offchainreporting2"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"github.com/smartcontractkit/sqlx"
+)
+
+type Relayer struct {
+	db       *sqlx.DB
+	chainSet evm.ChainSet
+	lggr     logger.Logger
+}
+
+func NewRelayer(db *sqlx.DB, chainSet evm.ChainSet, lggr logger.Logger) *Relayer {
+	return &Relayer{
+		db:       db,
+		chainSet: chainSet,
+		lggr:     lggr,
+	}
+}
+
+// No subservices started on relay start, but when the first job is started
+func (r *Relayer) Start() error {
+	return nil
+}
+
+// No peristent subservices to close on relay close
+func (r *Relayer) Close() error {
+	return nil
+}
+
+// Always ready
+func (r *Relayer) Ready() error {
+	return nil
+}
+
+// Always healthy
+func (r *Relayer) Healthy() error {
+	return nil
+}
+
+func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	spec, ok := s.(OCR2Spec)
+	if !ok {
+		return nil, errors.New("unsuccessful cast to 'ethereum.OCR2Spec'")
+	}
+	chain, err := r.chainSet.Get(spec.ChainID)
+	if err != nil {
+		return nil, err
```

Comment:
Wrap returned errors

Reply:
I don't think there's any additional ctx I want to wrap with here though, should be quite clear as is